### PR TITLE
[UPGRADE] Upgrade functions call that remove warnings

### PR DESCRIPTION
**Pull request by GuillaumeMilan**
Issue #7

### CODE REVIEW

File: lib/gitex.ex [hunk 1]
```diff
@@ -40,7 +40,7 @@ defmodule Gitex do
   def get_hash(ref,repo,path), do: get_hash(get(ref,repo),repo,path)
 
   def get_hash(hash,tree,repo,path), do: 
-    get_hash_path(hash,tree,repo,path |> String.strip(?/) |>  String.split("/"))
+    get_hash_path(hash,tree,repo,path |> String.trim("?") |>  String.split("/"))
```
Comment (Shakadak) [inline lib/gitex.ex]: Here you'll need to replace `"?"` with `"/"`

File: lib/gitex.ex [hunk 2]
```diff
@@ -51,7 +51,7 @@ defmodule Gitex do
   def put(%{tree: tree},repo,path,elem), do: put(object(tree,repo),repo,path,elem)
   def put(%{object: ref},repo,path,elem), do: put(object(ref,repo),repo,path,elem)
   def put(tree,repo,path,elem) when is_list(tree), do:
-    ({:tree,ref}=put_path(tree,repo,path |> String.strip(?/) |>  String.split("/"),elem); ref)
+    ({:tree,ref}=put_path(tree,repo,path |> String.trim("?") |>  String.split("/"),elem); ref)
```
Comment (Shakadak) [inline lib/gitex.ex]: Same stuff here, replace `"?"` with `"/"`

File: lib/standard_git.ex [hunk 3]
```diff
@@ -239,8 +239,8 @@ defmodule Gitex.Git do
     defp apply_delta_hunks(acc,base,<<0::size(1),add_len::size(7),add::binary-size(add_len)>> <> delta), do:
       apply_delta_hunks([acc,add],base,delta)
     defp apply_delta_hunks(acc,base,<<1::size(1),len_shift::bitstring-size(3),off_shift::bitstring-size(4)>> <> delta) do
-      off_ops = Enum.zip([0,8,16,24],Enum.reverse(for(<<x::size(1)<-off_shift>>,do: x))) |> Enum.filter_map(& elem(&1,1)==1,& elem(&1,0))
-      len_ops = Enum.zip([0,8,16],Enum.reverse(for(<<x::size(1)<-len_shift>>,do: x))) |> Enum.filter_map(& elem(&1,1)==1,& elem(&1,0))
+      off_ops = Enum.zip([0,8,16,24],Enum.reverse(for(<<x::size(1)<-off_shift>>,do: x))) |> Stream.filter(& elem(&1,1)==1) |> Enum.map(& elem(&1,0))
```
Comment (Shakadak) [inline lib/standard_git.ex:242]: Is the Stream usage necessary ? There is overhead in the construction and evaluation of a `Stream`, and I prefer to use Enum unless there is a tangible benefit.

File: mix.exs [hunk 4]
```diff
@@ -5,18 +5,20 @@ defmodule Gitex.Mixfile do
...
+        app: :gitex,
        readme: "README.md", main: "README",
        source_url: "https://github.com/awetzel/gitex",
        source_ref: "master"
      ],
      package: package(),
      description: description(),
-     deps: [{:ex_doc, ">= 0.0.0", only: :dev}]]
+     elixirc_options: [warnings_as_errors: true],
+     deps: [{:ex_doc, "~> 0.19", only: :dev, runtime: false}]]
```
Comment (Shakadak) [inline mix.exs:17]: Any reason for 0.19 instead of a more up to date version ?

File: mix.exs [hunk 5]
```diff
@@ -5,18 +5,20 @@ defmodule Gitex.Mixfile do
...
   end
 
   def application do
-    [applications: [:logger],
+    [extra_applications: [:crypto],
```
Comment (Shakadak) [inline mix.exs:21]: Can you  explain why `:crypto` is needed but not `:logger` in the extra_applications

**Review by Shakadak — CHANGES_REQUESTED** [review]
Substitutions of `"?"` by `"/"` are necessary, others comments can be discussed.

File: mix.exs (hunk 5 above)
Comment (GuillaumeMilan) [reply mix.exs:21]: The application depends on the crypto application as it uses some of its functions.
But the application doesn't use any logger any where. 
A good addition would be to log some stuff too.

File: mix.exs (hunk 4 above)
Comment (GuillaumeMilan) [reply mix.exs:17]: How sweet. It was change automatically by something but I can't figure out what.
If fix this too

File: lib/standard_git.ex (hunk 3 above)
Comment (GuillaumeMilan) [reply lib/standard_git.ex:242]: I don't know

**Pull request merged**